### PR TITLE
Rework the code of GPU selection

### PR DESCRIPTION
**Pull request by Mr-Milk**
Using official wrapper `pynvml` to acquire GPU information instead of reading from the command output.

### CODE REVIEW

**Comment by ArtemSokolov** [comment]
@clarenceyapp Do you have a multi-GPU machine that you can test this on? I will probably also need help with running a test on O2, because my FairShare is in the gutter right now due launching too many jobs over the past week.

**Comment by clarenceyapp** [comment]
@ArtemSokolov I do. 2x 3070s. Will report back.

**Comment by ArtemSokolov** [comment]
@clarenceyapp I verified that the new code work on O2.

**Comment by clarenceyapp** [comment]
@ArtemSokolov I confirm that the new code works on my local dual GPU machine.

File: toolbox/GPUselect.py [hunk 1]
```diff
@@ -1,19 +1,40 @@
...
+#     result=[]
+#     for row in gpu_output.split("\n"):
+#         m = memory_regex.search(row)
+#         if not m:
+#             continue
+#         gpu_memory = int(m.group("gpu_memory"))
+#         result.append(gpu_memory)
+#     return np.argsort(result)[0]
+
+
```
[the remark this answers is not in the document]
Comment (Mr-Milk) [reply toolbox/GPUselect.py]: Sure, I just delete it.

**Comment by ArtemSokolov** [comment]
Perfect. Thanks very much for contributing this, @Mr-Milk.

**Pull request merged**